src/core/services/theming/theming.js

```diff
@@ -528,7 +528,7 @@ function generateAllThemes($injector) {
   if (generateOnDemand) return;
 
   angular.forEach(THEMES, function(theme) {
-    if (!GENERATED[theme.name]) {
+    if (!GENERATED[theme.name] && !(theme.name !== $mdTheming.defaultTheme() && theme.name === 'default')) {
```

Comment:
I'm trying to figure when your addition will be valid,
```js
!(theme.name !== $mdTheming.defaultTheme() && theme.name === 'default')
```
let's say `theme.name !== $mdTheming.defaultTheme()` is true, than for sure `theme.name === 'default'` is false. - invalid
`theme.name !== $mdTheming.defaultTheme()` is false and `theme.name === 'default'` is false (or true) - invalid

it will only be valid if both are `true` which means `$mdTheming.defaultTheme() !== 'default'`